cylc/uiserver/handlers.py

```diff
@@ -314,32 +313,39 @@ class UIServerGraphQLHandler(CylcAppHandler, TornadoGraphQLHandler):
     def set_default_headers(self) -> None:
         self.set_header('Server', '')
 
-    def initialize(self, schema=None, executor=None, middleware=None,
-                   root_value=None, graphiql=False, pretty=False,
-                   batch=False, backend=None, auth=None, **kwargs):
+    def initialize(
+        self, schema=None, middleware=None, root_value=None, graphiql=False,
+        pretty=False, batch=False, execution_context_class=None,
+        validation_rules=None, auth=None, subscription_path=None, **kwargs
+    ):
         super(TornadoGraphQLHandler, self).initialize()
-        self.auth = auth
-        self.schema = schema
 
+        self.schema = schema or self.schema
```

Comment:
`or self.schema`?

(also seen in other variables below)

Is this to handle `initialize` being called on the same handler multiple times with different arguments?

Reply:
I think, if not set, it will pick up the defaults set by initialized `TornadoGraphQLHandler` parent..

Tornado spins up a new handler for each request I think.. But I think the options are set in the app on UIS startup..

Reply:
> Tornado spins up a new handler for each request I think

Yes. We rely on this for security reasons which is why this caught my interest.

Will have a quick check to make sure.